### PR TITLE
More `genesis-tool` operations & refactoring

### PR DESCRIPTION
**Pull request by deepfire**
1. Restructure `genesis-tool` in accord with https://github.com/input-output-hk/cardano-node/issues/84
2. New subcommands:  `keygen`, `to-verification`,  `redelegate` and `check-delegation`.
3. Redo exception handling..
4. Fix some warnings globally.
5. Small cleanups & indentation.

**NOTE: This PR includes file movement and reindentation, so it's split into relatively many commits -- I intend to weld them after review.**

### CODE REVIEW

File: cardano-cli/README.md [hunk 1]
```diff
@@ -58,39 +69,67 @@ The directory will have the following content:
...
-$ cabal new-run -- genesis-tool byron-pbft migrate-delegate-key-from byron-legacy \
-                                           --from key0.sk --to key0.pbft
+$ cabal new-run -- cardano-cli byron-pbft migrate-delegate-key-from byron-legacy \
+                                          --from key0.sk --to key0.pbft
 ```
 
+## Delegation
+
+The `redelegate` subcommand enables generation of delegation certificates,
+given the following inuputs:
```
Comment (ruhatch) [inline cardano-cli/README.md]:
```suggestion
given the following inputs:
```

File: cardano-cli/README.md [hunk 2]
```diff
@@ -58,39 +69,67 @@ The directory will have the following content:
...
+
+   - protocol magic
+   - starting epoch of delegation
+   - delegator signing key
+   - delegate verification key
+
+To check the generated delegation certificate, you can use the `check-delegation` subcommand,
+which would verify:
+
+   - certificatesignature validity
```
Comment (ruhatch) [inline cardano-cli/README.md]:
```suggestion
   - certificate signature validity
```

File: cardano-cli/README.md (hunk 1 above)
Comment (deepfire) [reply cardano-cli/README.md]: Thanks, fixed!

File: cardano-cli/README.md (hunk 2 above)
Comment (deepfire) [reply cardano-cli/README.md]: Fixed, thanks!

**Pull request merged**